### ISSUE TITLE
[Rule] Go: replace looppointer rule with exportloopref

### ISSUE THREAD

**Issue by arieltorti** [returntocorp/sgrep-rules#2385]
**Describe the bug**
The rule https://semgrep.dev/r?q=go.lang.correctness.looppointer.exported_loop_pointer&lang=Go is producing false positives.

It's my understanding that the rule should detect cases where the address of the loop variable is used or when the variable is exported (i.e: a closure is created) but it also detects a lot of safe cases as troublesome.

**To Reproduce**
```go
func() {
    values := []string{"a", "b", "c"}

    for _, val := range values {
        fmt.Println(val)
    }
}
```

**Expected behavior**
The rule shouldn't trigger. 

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Additional Context**
I think the rule should be refined to only trigger in cases where the loop variable address is used or when it's value is bind to a closure, but not when we're synchronously looping pointers.

**Comment by kurt-r2c** [comment]
Note - reporter is asking for behavior closer to https://github.com/kyoh86/exportloopref
Original rule is based on known-noisier behavior of https://github.com/kyoh86/looppointer

Remediation should be "add rule for exportloopref" and replace this rule with exportloopref rule in rule packs (if it exists)

**Comment by arieltorti** [comment]
That's exactly what's needed. How could I help here ?

**Comment by LewisArdern** [comment]
@arieltorti This got unresolved for a very long time (Sorry about that), I would like to help you get this resolved, core questions:

- Can you write the rule?  
- Would you need us to write the rule?

Based on that we can resolve this and get a fix or a new rule in.

**Comment by dgutson** [comment]
CC @qequ

**Comment by dgutson** [comment]
@LewisArdern currently @arieltorti and @qequ are focused on other parts of the project, so it is unlikely that we will be able to address this before 4 months minimum, so if you can write the rule, that would be great.
